File: benchmarks/beginner/B1-fibonacci/csharp/Program.cs
using System;
using System.Diagnostics;

const int N = 40;
const int RUNS = 10;
double[] timings = new double[RUNS];

Console.WriteLine("=== BENCHMARK: B1 - Fibonacci ===");
Console.WriteLine("Language: C#");
Console.WriteLine($"Runtime:  .NET {Environment.Version} (.NET 10)");
Console.WriteLine();

int result = 0;
for (int i = 0; i < RUNS; i++)
{
    Stopwatch sw = Stopwatch.StartNew();
    result = Fib(N);
    sw.Stop();
    double elapsed = sw.Elapsed.TotalMilliseconds;
    timings[i] = elapsed;
    Console.WriteLine($"Run {i + 1,2}: {elapsed:F2} ms");
}

Console.WriteLine();
double min = timings[0], max = timings[0], sum = 0;
foreach (double t in timings)
{
    sum += t;
    if (t < min)
    {
        min = t;
    }

    if (t > max)
    {
        max = t;
    }
}
double avg = sum / RUNS;
double variance = 0;
foreach (double t in timings)
{
    double diff = t - avg;
    variance += diff * diff;
}
double stddev = Math.Sqrt(variance / RUNS);

Console.WriteLine($"Min:    {min:F2} ms");
Console.WriteLine($"Avg:    {avg:F2} ms");
Console.WriteLine($"Max:    {max:F2} ms");
Console.WriteLine($"StdDev: {stddev:F2} ms");
Console.WriteLine();
string status = result == 102334155 ? "✓" : "✗";
Console.WriteLine($"Verification: {result} {status}");
return;

static int Fib(int n)
{
    if (n <= 1) return n;
    return Fib(n - 1) + Fib(n - 2);
}

File: benchmarks/beginner/B1-fibonacci/fsharp/Program.fs
open System
open System.Diagnostics

let rec fib n =
    if n <= 1 then n
    else fib (n - 1) + fib (n - 2)

let N = 40
let RUNS = 10
let timings = Array.zeroCreate<float> RUNS

printfn "=== BENCHMARK: B1 - Fibonacci ==="
printfn "Language: F#"
printfn $"Runtime:  .NET {Environment.Version} (.NET 10)"
printfn ""

let mutable result = 0
for i in 0 .. RUNS - 1 do
    let sw = Stopwatch.StartNew()
    result <- fib N
    sw.Stop()
    let elapsed = sw.Elapsed.TotalMilliseconds
    timings[i] <- elapsed
    printfn $"Run %2d{i + 1}: %.2f{elapsed} ms"

printfn ""
let minT = Array.min timings
let maxT = Array.max timings
let avg = Array.average timings
let variance = timings |> Array.averageBy (fun t -> (t - avg) ** 2.0)
let stddev = sqrt variance

printfn $"Min:    %.2f{minT} ms"
printfn $"Avg:    %.2f{avg} ms"
printfn $"Max:    %.2f{maxT} ms"
printfn $"StdDev: %.2f{stddev} ms"
printfn ""
let status = if result = 102334155 then "✓" else "✗"
printfn $"Verification: {result} {status}"
